s3stream/src/main/java/com/automq/stream/s3/wal/impl/object/ObjectWALService.java

```diff
@@ -84,10 +86,33 @@ public WALMetadata metadata() {
 
     @Override
     public AppendResult append(TraceContext context, ByteBuf data, int crc) throws OverCapacityException {
+        ByteBuf header = byteBufPool.get().retainedDuplicate();
+        final CompletableFuture<AppendResult.CallbackResult> appendResultFuture = new CompletableFuture<>();
+        appendResultFuture.whenComplete((result, cause) -> {
+            // Release the header buffer if it is not write to object storage.
+            if (header.refCnt() > 1) {
+                header.release();
+            }
+
+            // Return the header buffer to the buffer pool.
+            if (header.refCnt() == 1) {
+                byteBufPool.release(header);
+            } else {
+                log.error("[Bug] The header buffer is already released.");
+            }
+        });
```

Comment:
I think directly checking the `refCnt` of `ByteBuf` is not appropriate. We should ensure that the `ByteBuf` is correctly released through well-structured code, such as handling various exceptions.